kernel32/kernel32_windows.go

```diff
@@ -0,0 +1,75 @@
+package kernel32
+
+import (
+	"errors"
+	"github.com/mxplusb/windows"
+	"github.com/mxplusb/windows/common"
+)
+
+// GetComputerName retrieves the NetBIOS name of the local computer. This name is established at system startup, when the system reads it from the registry. See: https://msdn.microsoft.com/en-us/library/windows/desktop/ms724295(v=vs.85).aspx
+func GetComputerName() (string, error) {
+	var size windows.LpdWord = MaxComputerNameLength + 1
+	localBuffer := make([]windows.LptStr, size)
+	if err := getComputerName(&localBuffer[0], &size); err != nil {
+		return "", err
+	}
+	return common.LptStrToString(localBuffer), nil
+}
+
+type BackgroundProcess struct {
```

Comment:
Not quite sure of the intent of the "BackgroundProcess" here. Ultimately, the model for using ConPTY should be:

1. Create pair of anonymous pipes (anon-pipes are unidirectional)
2. [Create ConPTY, passing pipes created in 1](https://github.com/Microsoft/console/blob/master/samples/ConPTY/EchoCon/EchoCon/EchoCon.cpp#L32
3. Create a [StartupInfo struct, with a ProcThreadAttributeList PROC_THREAD_ATTRIBUTE_PSEUDOCONSOLE initialized to the hPTY returned in 2](https://microsoft.visualstudio.com/OS/_queries/query/877c336a-5a64-4fe6-a3d4-a7fce5376f70/)
4. Launch required command-line shell/app/tool/etc. [via call to CreateProcess(...), passing in the StartupInfoStruct created in 3](https://github.com/Microsoft/console/blob/master/samples/ConPTY/EchoCon/EchoCon/EchoCon.cpp#L45-L57)

After this, it's largely just a matter of squirting chars & VT to the command-line app, and listening for text/VT responses.

Reply:
`BackgroundProcess` and ConPTY aren't related, I just happened to have them on the same branch when I did the work since no one else currently works on this but me. Should I have been better, absolutely, was I when no one is holding me accountable, nah.

The reality is I currently don't have anonymous pipe support, and even if I did, if someone were to try to consume this and they weren't on 1809, there's no callout or safeguard in my library that says "hey this is version locked to 1809 so please don't use it otherwise." There is a lot of work to be done to make sure stuff like this is much more sustainable and safe for production work. Hard to do alone and without that sweet, sweet Microsoft engineering support.

That reminds me, I should come up with a todo so it can be more clearly defined what some of the goals and plans are...

Reply:
Boom. https://github.com/mxplusb/windows/projects/1